rpc/server.go

```diff
@@ -358,14 +475,45 @@ func (s *Server) exec(ctx context.Context, codec ServerCodec, req *serverRequest
 // execBatch executes the given requests and writes the result back using the codec.
 // It will only write the response back when the last request is processed.
 func (s *Server) execBatch(ctx context.Context, codec ServerCodec, requests []*serverRequest) {
+	lctx := ctx
+	s.pendingMu.Lock()
+	codecPending := s.pending[codec]
+	var reqIDs []interface{}
+	var preq *pendingRequest
+	if codecPending != nil {
+		reqIDs = make([]interface{}, len(requests))
+		var cancelFn func()
+		lctx, cancelFn = context.WithCancel(ctx)
+		preq = &pendingRequest{cancelFn, false}
+		for i, req := range requests {
+			reqIDs[i], _ = parseReqId(*req.id.(*json.RawMessage))
```

Comment:
I can not because I need it actually parsed, not only checked. I use it as a key for the "pending requests" map and it has to match the one received as a parameter of `rpc_cancel`. But I can move the type casting inside `parseReqId` and get rid of json in the server. Would that be ok?